pkg/platform/packet/packet.go

```diff
@@ -249,10 +283,12 @@ func (c *config) GetExpectedNodes() int {
 
 // checkValidConfig validates cluster configuration.
 func (c *config) checkValidConfig() hcl.Diagnostics {
+	// TODO: this function really needs to be tested
```

Comment:
I'm not sure if this comment make sense.

Also, maybe add the tests then? :wink:

Reply:
Yes, that was my internal TODO to add tests, not intended to be merged with that TODO. Sorry for the confusion :)

Test will be added, as mentioned in the PR description :)